### PR TITLE
Add support for base and target trees in git ingest, add .tar.gz bundler

### PR DESCRIPTION
**Pull request by evankanderson**
# Summary

Prepare for running external tools via Minder rego.  This adds two features (sorry, I got on a roll!):

* `git` ingest during a pull request can access the PR base branch (what you want to merge to) via `base_file.*` analogues of `file.*` operations.
* Added a `file.bundle` operation that takes a list of file paths and produces a `.tar.gz` as a string.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

Unit testing for `lib` and some light manual testing for the `git` ingest.

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [x] Checked that related changes are merged.

### CODE REVIEW

File: internal/engine/eval/rego/lib.go
Comment (JAORMX) [inline internal/engine/eval/rego/lib.go:450]: Do we want to expose this function to non PR/diff contexts too? I think it would be handy in general.

File: internal/engine/ingester/git/git.go
Comment (JAORMX) [inline internal/engine/ingester/git/git.go:131]: We don't need to fix this in this PR... But... I wonder if we should have a base and target checkpoints checksum checkpoints?

File: internal/engine/eval/rego/lib.go
Comment (evankanderson) [reply internal/engine/eval/rego/lib.go:453]: Talking with Ed, he suggested `archive` instead.

File: internal/engine/eval/rego/lib.go
Comment (evankanderson) [reply internal/engine/eval/rego/lib.go:450]: See `FileArchive`, above.  😁

File: internal/engine/ingester/git/git.go
Comment (evankanderson) [reply internal/engine/ingester/git/git.go:131]: Since "base" is the branch that the PR will be merged into, I think that only changes if the target/head branch changes.

**Review by JAORMX — DISMISSED** [review]

**Review by JAORMX — APPROVED** [review]

**Comment by evankanderson** [comment]
I'm going to put these behind flags, then it should be ready to merge.

**Comment by evankanderson** [comment]
Okay, I think this is ready to merge, assuming that the tests pass.  I've put both the new functions and the new ingest type behind flags, and corrected the flag check on UpdateRuleType to block the pr templates (it was only on create previously).

**Review by JAORMX — APPROVED** [review]

**Pull request merged**
